Runtime/Data/Type/Transformation/CollectionAggregator.cs

```diff
@@ -23,6 +23,23 @@ public abstract class CollectionAggregator<TInput, TOutput, TEvent, TCollection,
         [field: DocumentedByXml]
         public TCollection Collection { get; set; }
 
+        /// <summary>
+        /// Processes the current collection and sums the collection total.
+        /// </summary>
+        /// <returns>The summed value of all values in the collection.</returns>
+        public virtual TOutput Transform()
+        {
+            return Transform(Collection.NonSubscribableElements[Collection.CurrentIndex]);
```

Comment:
Just at a single index? Either this is wrong or the XML docs for this method (and `DoTransform`) should explain that this only operates on the `collection's current index` or something.

Reply:
This is an implementation detail, it isn't required public knowledge about how it's performing the aggregation on the internal collection.

And the way it does that is to simply call `Transform(ObjectAtCurrentIndex)` 

which internally goes to the `Transform(Object)` method, which sets the given `Object` to the index of the collection at the `CurrentIndex`

So doing:

```
CurrentIndex = 1;
Transform(Object);

>>> does

Collection[CurrentIndex] = Object
ProcessCollection()
```

is the same as doing

```
CurrentIndex = 1
Transform()

>>> does

Transform()

Transform(ObjectAtCurrentIndex);
Collection[CurrentIndex] = Object
ProcessCollection()
```